spring-reactive-modules/spring-reactive-client-2/src/test/java/com/baeldung/fluxmapandondonext/DoOnNextUnitTest.java

```diff
@@ -0,0 +1,51 @@
+package com.baeldung.fluxmapandondonext;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.test.StepVerifier;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+class DoOnNextUnitTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DoOnNextUnitTest.class);
+
+    @Test
+    void givenFluxOfNumbers_whenCountingNumbers_thenReturnCorrectCountAndLogToOutput() {
+        AtomicInteger counter = new AtomicInteger(0);
+
+        Flux<Integer> numberFlux = Flux.just(1, 2, 3, 4, 5)
+            .doOnNext(number -> {
+                LOGGER.info(String.valueOf(number));
+                counter.incrementAndGet();
+            })
+            .onErrorResume(Flux::error);
+
+        StepVerifier.create(numberFlux)
+            .expectNextCount(5)
+            .verifyComplete();
+
+        Assertions.assertEquals(counter.get(), 5);
```

Comment:
Let's `static import` the `assertEquals` function here. Also, note that the expected is the first parameter in JUnit assertions, not the second as you have it here.

Reply:
Done.
Thanks a lot.